library/core/class.controller.php

```diff
@@ -2142,4 +2164,23 @@ public function title($Title = null, $Subtitle = null) {
 
         return $this->data('Title');
     }
+
+    /**
+     * Set the destination URL where the page will be redirected after an ajax request.
+     *
+     * @param string|null $destination Destination URL or path.
+     *      Redirect to current URL if nothing or null is supplied.
+     * @param bool $trustedOnly Non trusted destinations will be redirected to /home/leaving?Target=$destination
+     */
+    public function redirectTo($destination = null, $trustedOnly = true) {
+        if ($destination === null) {
+            $url = url('');
+        } elseif ($trustedOnly) {
+            $url = safeURL($destination);
+        } else {
+            $url = url($destination);
+        }
+
+        $this->redirectTo = $url;
```

Comment:
If it *sets* the `RedirectUrl`, it's good. But what if it's hooking into an event and is trying to *read* the `RedirectUrl` property that was previously set in any of these dozens of locations?